api/src/clojure/org/akvo/flow_api/datastore/survey.clj

```diff
@@ -59,6 +60,7 @@
                  survey
                  (anomaly/not-found "Survey not found"
                                     {:survey-id survey-id}))
+        registration-form-id (.getNewLocaleSurveyId survey)
```

Comment:
In the case the survey isn't found, an exception is thrown in the previous line right?